frontend/src/app/main/ui/workspace/context_menu.cljs

```diff
@@ -33,8 +34,33 @@
   (dom/prevent-default event)
   (dom/stop-propagation event))
 
+
+(mf/defc element-icon
+  [{:keys [shape] :as props}]
+  (case (:type shape)
+    :frame i/artboard
+    :image i/image
+    :line i/line
+    :circle i/circle
+    :path i/curve
+    :rect i/box
+    :text i/text
+    :group (if (some? (:component-id shape))
+             i/component
+             (if (:masked-group? shape)
+               i/mask
+               i/folder))
+    :bool (case (:bool-type shape)
+            :difference   i/bool-difference
+            :exclude      i/bool-exclude
+            :intersection i/bool-intersection
+            #_:default    i/bool-union)
+    :svg-raw i/file-svg
+    nil))
```

Comment:
I think this could be a component in main/ui/components named shape-icon. This way is not duplicated and is clearer to import it from layers menu and sidebars.

Reply:
Great idea¡¡